src/posts/2020-08-20-an-indepth-guide-to-aria-roles.md

```diff
@@ -0,0 +1,190 @@
+  </div>
+</div>
+```
+
+#### Tooltip example
+
+```html
+<button aria-describedby="notifications-desc">Notifications</button>  
+<div role="tooltip" id="notifications-desc">View and manage notifications</div>  
+```
+
+- View the [full list of available Document Structure roles](https://www.w3.org/TR/wai-aria/#document_structure_roles).
+
+### Landmark Roles
+
+Landmark roles identify content areas within a page, and as the name implies serve as navigational landmarks for the types of content areas that users may be interested in. These content areas include the page's primary content, collections of navigation links, and search functions to name a few.
+
+Most Landmark roles are also implicitly exposed via specific Grouping and Sectioning HTML elements. For example:
+
+- The `banner` role can be associated with the `<header>` element
+- The `complementary` role can be associated with the `<aside>` element
+- The `contentinfo` role can be associated with the `<footer>` element
+
+Refer to [Quick Tip: ARIA Landmark roles and HTML5 implicit mapping](https://www.a11yproject.com/posts/2013-01-14-aria-landmark-roles/#toc_HTML5-implicit-mappings-of-Landmark-roles) for more details on the above mappings and browser support for them.
+
+#### Navigation role example
+
+```html
+<nav role="navigation">
+  <ul>
+    <li><a href="#a">Dexter</a></li>
+    <li><a href="#b">Doctor Who</a></li>
+    <li><a href="#c">Futurama</a></li>
+  </ul>  
+</nav>
+```
+
+#### Search role example
+
+```html
+<form role="search">
+  <label for="search-input">Search this site</label>
+  <input type="text" id="search-input" name="search">
+  <input type="submit" name="submit-btn" value="Search" />
+</form>
+```
+
+#### Footer role example
+
+```html
+<footer role="contentinfo">
+  <p>&copy; 2020 Small Business Ltd. All rights reserved.</p>
+</footer>
+```
+
+- View the [full list of available Landmark roles](https://www.w3.org/TR/wai-aria/#landmark_roles).
+
+### Live Region Roles
+
+Live Region roles are used to inform assistive technologies, such as screen readers, of dynamic changes to page content, so that they can be announced to users. The reason for this is that dynamic changes may be visually noticeable, but likely will be missed by users with low or no vision.
+
+#### Alert role example
+
+The most commonly used Live Region role is the `alert` role.
+
+```html
+<div role="alert">
+  <p>Autosave failed.</p>
+</div>
+```
+
+When using the `alert` role, keep in mind that it should not be a focusable element and it should not require users to close it.
+
+- View the [full list of available Live Region roles](https://www.w3.org/TR/wai-aria/#live_region_roles).
+
+### Widget Roles
+
+Widget roles are used to describe common interactive patterns or those that currently lack semantic equivalents in HTML. In most cases, these interactive patterns will be JavaScript-driven. Widget roles should not be confused with `role="widget"` which is itself an abstract role.
```

Comment:
I'd suggest linking to [Abtract Roles in the spec](https://www.w3.org/TR/wai-aria-1.1/#isAbstract) to help inform the reader about what they are and why they shouldn't be used in code.